kwok/entrypoint.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+show_info() {
+  sleep 1
+
+  # Print some useful information
+  echo "###############################################################################"
+  echo "> kubectl -s :${KWOK_KUBE_APISERVER_PORT} version"
+  kwokctl kubectl version || true
+
+  echo "###############################################################################"
+  echo "# The following kubeconfig can be used to connect to the Kubernetes API server"
+  cat <<EOF
+apiVersion: v1
+clusters:
+- cluster:
+    server: http://127.0.0.1:${KWOK_KUBE_APISERVER_PORT}
+  name: kwok
+contexts:
+- context:
+    cluster: kwok
+  name: kwok
+current-context: kwok
+kind: Config
+preferences: {}
+users: null
+EOF
+
+  echo "###############################################################################"
+  echo "> kubectl -s :${KWOK_KUBE_APISERVER_PORT} get ns"
+  kwokctl kubectl get ns || true
+
+  echo "###############################################################################"
+  echo "# The above example works if your host's port is the same as the container's,"
+  echo "# otherwise, change it to your host's port"
+}
+
+# Create a cluster
+KWOK_KUBE_APISERVER_PORT=0 kwokctl create cluster "$@" || exit 1
+
+show_info &
+
+kwokctl kubectl apply -f toolchain.dev.openshift.com_usersignups.yaml && sh ./user-signups.sh
```

Comment:
since you are running user-signups.sh from the entry point script you are essentially loading the data during image startup instead of during the build. We want to do this during the build.

Reply:
we don't have a preexisting cluster from the image. the cluster itself is created during run time, applying this during the build process won't work

Reply:
My aim is to create the cluster at build time. I can see `kwok` as `start` and `stop` commands, so we can probably create at build time, stop and then start it back when the image starts.

Reply:
Done